src/cargo/ops/cargo_new.rs

```diff
@@ -797,3 +791,40 @@ fn discover_author() -> CargoResult<(String, Option<String>)> {
 
     Ok((name, email))
 }
+
+fn find_git_config() -> Option<GitConfig> {
+    let inside_tests = env::var("__CARGO_TEST_ROOT");
+
+    if let Ok(test_root) = inside_tests {
+        find_tests_git_config(test_root)
+    } else {
+        find_real_git_config()
+    }
```

Comment:
This can also be simplified.

```suggestion
    match env::var("__CARGO_TEST_ROOT") {
        Ok(test_root) => find_tests_git_config(test_root),
        Err(_) => find_real_git_config(),
    }
```